composition-js/src/merging/merge.ts

```diff
@@ -1810,79 +1848,54 @@ class Merger {
     //   definition is the intersection of all definitions (meaning that if there divergence in
     //   locations, we only expose locations that are common everywhere).
     this.mergeDescription(sources, dest);
-    if (sources.some((s) => s && this.isMergedDirective(s))) {
+    if (sources.some((s) => s && s.locations.some(loc => executableDirectiveLocations.includes(loc)))) {
       this.mergeExecutableDirectiveDefinition(sources, dest);
     }
   }
 
   // Note: as far as directive definition goes, we currently only merge directive having execution location, and only for
-  // thos locations. Any type system directive definition that propagates to the supergraph (graphQL built-ins and `@tag`)
+  // those locations. Any type system directive definition that propagates to the supergraph (graphQL built-ins and `@tag`)
   // is currently handled in an hard-coded way. This will change very soon however so keeping this code around to be
   // re-enabled by a future commit.
-  //private mergeTypeSystemDirectiveDefinition(sources: (DirectiveDefinition | undefined)[], dest: DirectiveDefinition) {
```

Comment:
As we re-enable this method, it would be nice to re-enable the currently skipped related tests in `hints.test.ts`.

Reply:
I don't think that the tests as written are applicable since we only do this detection when the directive is explicitly exposed.

Reply:
While it's fair that we cannot just remove the `skip` on those test and them work, the spirit of my comment that:
1. there is no point in keeping those skipped test around as skipped after this patch.
2. we really should have test for the hints enabled by this patch.

I'd personally just reuse those tests (I'll note that you've removed one of them, but see my other comment for why I think it should be re-added) and modify them ever so slightly to use a manually exposed random directive, but any alternatives that achieve the 2 points above wfm too.